src/ucp/tag/offload.c

```diff
@@ -276,6 +276,10 @@ ucp_tag_offload_do_post(ucp_request_t *req)
             return status;
         }
 
+        if (!req->recv.state.dt.contig.md_map) {
```

Comment:
Not needed. Per discussion in #4499 we have to fail `ucp_request_memory_reg` in case of memory mismatch to avoid such checks on every zcopy protocol.
Should be fixed by  #4499